corehq/apps/hqadmin/views.py

```diff
@@ -467,6 +468,11 @@ def get(self, request, *args, **kwargs):
         if not self.user:
             return HttpResponseNotFound('User %s not found.' % full_username)
 
+        if 'domain' in kwargs:
+            # domain check when this get's called as domain level admin restore.
```

Comment:
Feels weird to have a check for the subclass be here. What if you made a `validate_user` method here which checks all the user properties which you could then override in the subclass?